scaffold-api/src/scaffold_api/resources/apihelper.py

```diff
@@ -26,10 +31,70 @@ class Api(BaseApi):
     @property
     def specs_url(self):
         """Return URL for endpoint."""
-        scheme = 'http' if '5000' in self.base_url else 'https'
-        return url_for(self.endpoint('specs'), _external=True, _scheme=scheme)
+        scheme = "http" if "3200" in self.base_url else "https"
+        return url_for(self.endpoint("specs"), _external=True, _scheme=scheme)
+
+    @classmethod
+    def common_decorator(cls, api, endpoint_description):
+        """Common decorators for the resources"""
+
+        def decorator(func):
+            @wraps(func)
+            @cross_origin(origins=allowedorigins())
+            @auth.require
+            @api.doc(description=endpoint_description)
+            @api.response(401, "Unauthorized")
+            @api.response(500, "Internal Server Error")
+            def wrapper(*args, **kwargs):
+                return func(*args, **kwargs)
+
+            return wrapper
+
+        return decorator
+
+    @classmethod
+    def convert_ma_schema_to_restx_model(cls, api, schema, name):
```

Comment:
from Gemini:

from flask_restx import fields, Api

def convert_ma_schema_to_restx_model(api: Api, schema: ma.Schema, name: str) -> fields.Model:
    """Converts a Marshmallow schema to a Flask-RESTX model, handling nested fields.

    Args:
        api: The Flask-RESTX API instance.
        schema: The Marshmallow schema instance.
        name: The desired name for the Flask-RESTX model.

    Returns:
        A Flask-RESTX model with nested fields support.

    Raises:
        ValueError: If an unsupported Marshmallow field type is encountered.
    """

    model_fields = {}
    for field_name, field in schema.fields.items():
        field_type = type(field)

        # Basic field types
        if field_type in (ma_fields.Integer, ma_fields.String, ma_fields.Float,
                          ma_fields.Boolean, ma_fields.DateTime):
            model_fields[field_name] = _get_basic_field(field)

        # Nested fields
        elif field_type == ma.fields.Nested:
            nested_schema = field.schema
            model_fields[field_name] = fields.Nested(
                convert_ma_schema_to_restx_model(api, nested_schema, f"{name}_{field_name}")
            )

        else:
            raise ValueError(f"Unsupported Marshmallow field type: {field_type}")

    return api.model(name, model_fields)

def _get_basic_field(field: ma.fields.Field) -> fields.Field:
    """Creates a basic Flask-RESTX field from a Marshmallow field.

    Args:
        field: The Marshmallow field.

    Returns:
        A corresponding Flask-RESTX field.
    """

    return fields.Field(
        required=field.required,
        description=field.metadata.get("description", ""),
    )